tests/ui/parser/issues/issue-67377-invalid-syntax-in-enum-discriminant.stderr

```diff
@@ -1,47 +1,26 @@
-error: mismatched closing delimiter: `]`
-  --> $DIR/issue-67377-invalid-syntax-in-enum-discriminant.rs:5:27
-   |
-LL |         V = [PhantomData; { [ () ].len() ].len() as isize,
-   |             -             ^              ^ mismatched closing delimiter
-   |             |             |
-   |             |             unclosed delimiter
-   |             closing delimiter possibly meant for this
-
-error: mismatched closing delimiter: `]`
-  --> $DIR/issue-67377-invalid-syntax-in-enum-discriminant.rs:12:24
-   |
-LL |         V = [Vec::new; { [].len()  ].len() as isize,
-   |             -          ^           ^ mismatched closing delimiter
-   |             |          |
-   |             |          unclosed delimiter
-   |             closing delimiter possibly meant for this
-
-error: mismatched closing delimiter: `]`
-  --> $DIR/issue-67377-invalid-syntax-in-enum-discriminant.rs:19:24
-   |
-LL |         V = [Vec::new; { [0].len() ].len() as isize,
-   |             -          ^           ^ mismatched closing delimiter
-   |             |          |
-   |             |          unclosed delimiter
-   |             closing delimiter possibly meant for this
-
 error: this file contains an unclosed delimiter
-  --> $DIR/issue-67377-invalid-syntax-in-enum-discriminant.rs:23:65
+  --> $DIR/issue-67377-invalid-syntax-in-enum-discriminant.rs:20:65
    |
 LL |         V = [PhantomData; { [ () ].len() ].len() as isize,
-   |                                          - missing open `[` for this delimiter
+   |                           -              - missing open `[` for this delimiter
+   |                           |
+   |                           the last unclosed delimiter before unmatched delimiter
 ...
 LL |         V = [Vec::new; { [].len()  ].len() as isize,
-   |                                    - missing open `[` for this delimiter
+   |                        -           - missing open `[` for this delimiter
+   |                        |
+   |                        the last unclosed delimiter before unmatched delimiter
```

Comment:
it's more clear only report as `unclosed delimiter` as before?

this is the mismatched scenario, means we have the same number openning delimiter and closing delimiter, but they are just not matched, for example:

```
({)) we have 2 open delimiter and 2 close delimiter, but it's a mismatched pair here
[{[)}] we have 3 open delimiter and 3 close, but there is a mismatchd pair
``
so in mismatch scenario, seems we should only report mismatch as before.

only for the scenario, that openning delimiter and closing delimiter is not same, for example:

```
[)] here we have 1 open, and 2 close delimiter, it's safe to report missing a `(`.
```